CodePush.js

```diff
@@ -170,8 +183,31 @@ function setUpTestDependencies(testSdk, providedTestConfig, testNativeBridge) {
   if (testNativeBridge) NativeCodePush = testNativeBridge;
 }
 
+// This function allows only one syncInternal operation to proceed at any given time.
+// Parallel calls to sync() while one is ongoing yields CodePush.SyncStatus.SYNC_IN_PROGRESS.
+const sync = (() => {
+  let syncInProgress = false;
+  const setSyncCompleted = () => { syncInProgress = false; };
+  
+  return (options = {}, syncStatusChangeCallback, downloadProgressCallback) => {
```

Comment:
Could we do "return (...args) {" and then do "syncInternal(...args)" below? That way we don't have to define the signature/default values twice.